File: Behavioral/Strategy/strategy.py
class Strategy(ABC):
    @abstractmethod
    def do_algorithm(self, data): pass

class Context:
    def __init__(self, strategy: Strategy):
        self._strategy = strategy
    def set_strategy(self, strategy: Strategy):
        self._strategy = strategy
    def do_something(self):
        result = self._strategy.do_algorithm(["a", "b"])
        print(result)

File: Behavioral/Strategy/strategy.cs
public interface IStrategy { object DoAlgorithm(object data); }

class Context {
    private IStrategy _strategy;
    public void SetStrategy(IStrategy strategy) => _strategy = strategy;
    public void Execute() => _strategy.DoAlgorithm("Sorting data");
}

class ConcreteStrategyA : IStrategy {
    public object DoAlgorithm(object data) => $"Strategy A: {data}";
}